### PR TITLE
Fix per-year bugs in redeeming team admin access

### LINKED ISSUE #2633: Unable to Redeem Team Mod Code Two Years In A Row
This fails if you've already redeemed a code in a prior year.

It shouldn't...

### PR DESCRIPTION
**Pull request by phil-lopreiato**
No longer fail if the user has redeemed a team admin code for a prior year

## Description
We should be checking the years in here

## Motivation and Context
Fixes #2633 

## How Has This Been Tested?
added unit test

### CODE REVIEW

File: controllers/team_admin_controller.py [hunk 1]
```diff
@@ -83,7 +83,8 @@ def get(self):
             for team in teams_future
         }
         team_num_to_robot_name = {
-            int(robot.get_result().team.id()[3:]): robot.get_result().robot_name
+            int(robot.get_result().team.id()[3:]):
+            robot.get_result().robot_name
```
Comment (gregmarra) [inline controllers/team_admin_controller.py]: Should this be indented for style?

File: controllers/team_admin_controller.py [hunk 2]
```diff
@@ -112,14 +113,22 @@ def get(self):
...
         self.template_values.update({
-            "existing_access": existing_access,
-            "teams": team_num_to_team,
-            "robot_names_by_team": team_num_to_robot_name,
-            "team_medias": team_medias,
-            "team_social_medias": team_social_medias,
-            "suggestions_by_team": suggestions_by_team,
-            "suggestion_names": self.SUGGESTION_NAMES,
-            "suggestion_review_urls": self.SUGGESTION_REVIEW_URL,
+            "existing_access":
```
Comment (gregmarra) [inline controllers/team_admin_controller.py]: Why split these?

**Comment by gregmarra** [comment]
FIRST released 2020 codes in 2019. You're generating `TeamAdminAccess.year` from datetime.now(). Will that create any issues?

File: controllers/team_admin_controller.py (hunk 1 above)
Comment (phil-lopreiato) [reply controllers/team_admin_controller.py]: argh, stupid autoformat

File: controllers/team_admin_controller.py (hunk 2 above)
Comment (phil-lopreiato) [reply controllers/team_admin_controller.py]: more autoformat :(

**Comment by gregmarra** [comment]
I'd probably allow them to work on any year, and expire each year. You're
basically an authorized rep of that team's full history, but you shouldn't
be a rep 2 years from now.

**Comment by phil-lopreiato** [comment]
Updated to accept codes from any year, removes requirement that one account be linked to a team at any point in time

**Pull request merged**